### PR TITLE
Don't bother running lints if we're going to ignore them

### PR DESCRIPTION
**Pull request by gsnedders**
This presently has tests as any testing will likely need `mock` and some refactoring.

---

This change is [Reviewable](https://reviewable.io/reviews/w3c/wpt-tools/111)

### CODE REVIEW

**Comment by Ms2ger** [comment]
Perhaps `parse_whitelist` should return `ignored_files` separately?

**Comment by gsnedders** [comment]
Yeah, probably. Now done.

---

Review status: 0 of 6 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/wpt-tools/111#-:-KQvFwSL40v8o0GX5GxN:b4uo6n5)*

File: lint/lint.py [hunk 1]
```diff
@@ -398,6 +403,15 @@ def process_errors(path, errors):
         if not os.path.exists(abs_path):
             continue
 
+        ignored = False
+        for file_match in ignored_files:
```
Comment (Ms2ger) [inline lint/lint.py]: I think one would usually write this as `any(fnmatch.fnmatch(path, file_match) for file_match in ignored_files)`

File: lint/lint.py [hunk 2]
```diff
@@ -83,10 +83,9 @@ def filter_whitelist_errors(data, path, errors):
 
     for file_match, whitelist_errors in iteritems(data):
         if fnmatch.fnmatch(path, file_match):
+            assert "*" not in whitelist_errors
```
Comment (Ms2ger) [inline lint/lint.py]: Move this before the `if` (or remove it; either is fine by me).

**Comment by Ms2ger** [comment]
r+ on the code with those nits fixed; I don't feel comfortable reviewing the tests.

**Comment by gsnedders** [comment]
Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[lint/lint.py, line 407 \[r1\]](https://reviewable.io:443/reviews/w3c/wpt-tools/111#-KQuxGnpwIHk0oPEKHuY-r1-407:-KR-EC283drE0Gwca6md:b-896fix) ([raw file](https://github.com/w3c/wpt-tools/blob/8ba2f23f252809f55cb2a2c9dc382cef88840872/lint/lint.py#L407)):*
<details><summary><i>Previously, Ms2ger wrote…</i></summary>
> I think one would usually write this as `any(fnmatch.fnmatch(path, file_match) for file_match in ignored_files)`
</details>
Done.

---

*[lint/lint.py, line 86 \[r2\]](https://reviewable.io:443/reviews/w3c/wpt-tools/111#-KQuxGnpwIHk0oPEKHuY-r2-86:-KR-E8NnyrvXnk7KIvee:b-896fix) ([raw file](https://github.com/w3c/wpt-tools/blob/3b87a54f571a2221e4ab1cf8f06f2de6571b1960/lint/lint.py#L86)):*
<details><summary><i>Previously, Ms2ger wrote…</i></summary>
> Move this before the `if` (or remove it; either is fine by me).
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/wpt-tools/111)*

**Comment by jgraham** [comment]
Reviewed 1 of 1 files at r2, 5 of 5 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[lint/tests/test_lint.py, line 106 \[r3\]](https://reviewable.io:443/reviews/w3c/wpt-tools/111#-KR-MrM8_P432CgS0zih:-KR-MrMA7wA4UPdeM0oz:b2kfd9y) ([raw file](https://github.com/w3c/wpt-tools/blob/38bbf7148ac4abee7dc9b2d1f207e28d43f043fc/lint/tests/test_lint.py#L106)):*
> ```Python
> 
> @pytest.mark.skipif(six.PY3, reason="lint.lint doesn't support Python 3")
> def test_lint_for_realz(capsys):
> ```

s/for_realz/[some string that's actually English]/

---

*[lint/tests/test_lint.py, line 114 \[r3\]](https://reviewable.io:443/reviews/w3c/wpt-tools/111#-KR-Dl16eMat0Oiao1Fp:-KR-Dl18c1ZVNuEy82pP:b-s00z0w) ([raw file](https://github.com/w3c/wpt-tools/blob/38bbf7148ac4abee7dc9b2d1f207e28d43f043fc/lint/tests/test_lint.py#L114)):*
> ```Python
>             assert mocked_check_file_contents.call_count == 1
>     out, err = capsys.readouterr()
>     assert out == """
> ```

No, this isn't supposed to be a stable API. The json output is though, so you could test that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/wpt-tools/111)*

**Comment by gsnedders** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[lint/tests/test_lint.py, line 114 \[r3\]](https://reviewable.io:443/reviews/w3c/wpt-tools/111#-KR-Dl16eMat0Oiao1Fp:-KR-Ng-R5YgCwwweeNvT:b-111pwr) ([raw file](https://github.com/w3c/wpt-tools/blob/38bbf7148ac4abee7dc9b2d1f207e28d43f043fc/lint/tests/test_lint.py#L114)):*
<details><summary><i>Previously, jgraham wrote…</i></summary>
> No, this isn't supposed to be a stable API. The json output is though, so you could test that.
</details>
What *should* be tested here? That the output is non-empty? That the output includes "TRAILING WHITESPACE" and "broken.html"? Because we should definitely guarantee the unstructured output is useful.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/wpt-tools/111)*

**Comment by jgraham** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[lint/tests/test_lint.py, line 114 \[r3\]](https://reviewable.io:443/reviews/w3c/wpt-tools/111#-KR-Dl16eMat0Oiao1Fp:-KR-OWxV6th7VkmN9fI-:b-34rejb) ([raw file](https://github.com/w3c/wpt-tools/blob/38bbf7148ac4abee7dc9b2d1f207e28d43f043fc/lint/tests/test_lint.py#L114)):*
<details><summary><i>Previously, gsnedders (Geoffrey Sneddon) wrote…</i></summary>
> What *should* be tested here? That the output is non-empty? That the output includes "TRAILING WHITESPACE" and "broken.html"? Because we should definitely guarantee the unstructured output is useful.
</details>
I think a test here is pretty silly since it would be so trivial to notice if it was broken. But to the extent that it isn't silly it can't check more than the expected error code and the name of the file.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/wpt-tools/111)*

**Comment by gsnedders** [comment]
*[lint/tests/test_lint.py, line 114 \[r3\]](https://reviewable.io:443/reviews/w3c/wpt-tools/111#-KR-Dl16eMat0Oiao1Fp:-KR4Nw9YQ2qjYvfeHWBm:be2rf24) ([raw file](https://github.com/w3c/wpt-tools/blob/38bbf7148ac4abee7dc9b2d1f207e28d43f043fc/lint/tests/test_lint.py#L114)):*
<details><summary><i>Previously, jgraham wrote…</i></summary>
> I think a test here is pretty silly since it would be so trivial to notice if it was broken. But to the extent that it isn't silly it can't check more than the expected error code and the name of the file.
</details>
I'm less convinced by that argument. I can well imagine breaking this and nobody noticing until after we updated tools in w-p-t.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/wpt-tools/111)*

**Comment by gsnedders** [comment]
@jgraham ping

**Comment by jgraham** [comment]
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/wpt-tools/111)*

**Pull request merged**